src/Kiota.Builder/Writers/HTTP/CodeClassDeclarationWriter.cs

```diff
@@ -0,0 +1,387 @@
+        var pathParameters = requestBuilderClass
+            .GetChildElements(true)
+            .OfType<CodeProperty>()
+            .Where(property => property.IsOfKind(CodePropertyKind.PathParameters) && !property.Name.Equals(Constants.PathParameters, StringComparison.OrdinalIgnoreCase))
+            .ToArray();
+
+        return pathParameters;
+    }
+
+    /// <summary>
+    /// Retrieves the base URL for the given request builder class.
+    /// </summary>
+    /// <param name="requestBuilderClass">The request builder class containing the base URL property.</param>
+    /// <returns>The base URL as a string, or null if not found.</returns>
+    private static string? GetBaseUrl(CodeClass requestBuilderClass)
+    {
+        // Retrieve the base URL property from the request builder class
+        return requestBuilderClass.Properties
+            .FirstOrDefault(property => property.Name.Equals(Constants.BaseUrl, StringComparison.OrdinalIgnoreCase))?.DefaultValue;
+    }
+
+    /// <summary>
+    /// Retrieves the URL template property for the given request builder class.
+    /// </summary>
+    /// <param name="requestBuilderClass">The request builder class containing the URL template property.</param>
+    /// <returns>The URL template property, or null if not found.</returns>
+    private static CodeProperty? GetUrlTemplateProperty(CodeClass requestBuilderClass)
+    {
+        // Retrieve the URL template property from the request builder class
+        return requestBuilderClass
+            .GetChildElements(true)
+            .OfType<CodeProperty>()
+            .FirstOrDefault(static property => property.IsOfKind(CodePropertyKind.UrlTemplate));
+    }
+
+    /// <summary>
+    /// Writes the URL template for the given URL template property to the writer.
+    /// </summary>
+    /// <param name="urlTemplateProperty">The URL template property containing the URL template.</param>
+    /// <param name="writer">The language writer to write the URL template to.</param>
+    private static void WriteUrlTemplate(CodeProperty urlTemplateProperty, LanguageWriter writer)
+    {
+        // Write the URL template documentation as a comment
+        writer.WriteLine($"# {urlTemplateProperty.Documentation?.DescriptionTemplate}");
+
+        // Write the URL template value
+        writer.WriteLine($"# {urlTemplateProperty.DefaultValue}");
+
+        // Write an empty line for separation
+        writer.WriteLine();
+    }
+
+    /// <summary>
+    /// Writes the path parameters for the given request builder class to the writer.
+    /// </summary>
+    /// <param name="pathParameters">The array of path parameters to write.</param>
+    /// <param name="writer">The language writer to write the path parameters to.</param>
+    private static void WritePathParameters(CodeProperty[] pathParameters, LanguageWriter writer)
+    {
+        if (pathParameters is null) return;
+
```

Comment:
```suggestion
```

parameter cannot be null since method is private and we have NRT enabled on the project. Compiler would yell at you otherwise.